module/takeuntil/index.js

```diff
@@ -1,7 +1,12 @@
 var flyd = require('../../lib');
+var drop = require('ramda/src/drop');
 
-module.exports = flyd.curryN(2, function(src, term) {
-  return flyd.endsOn(flyd.merge(term, src.end), flyd.combine(function(src, self) {
+var drop1 = flyd.transduce(drop(1));
+
+module.exports = flyd.curryN(2, function(term, src) {
+  var end$ = flyd.merge(term.hasVal ? drop1(term) : term, src.end);
```

Comment:
To maintain the behaviour of takeUntil as it was before #180 we need to drop a single value from the terminator stream if it already has a value, otherwise the stream will end immediately.

Reply:
an example of code that wouldn't work without it would be:

```js
const s = stream();
const result$ = s.chain(q => fromPromise(getSearchResults(q)))
  .pipe(takeUntil(s))
  .map(results => h('ul', results.content.map(res => h('li', res))))
  .map(render('#results'));
```

In the above example you would only be able to search once. Any subsequent search would be ended immediately and not return a value.